src/engine/GroupBy.cpp

```diff
@@ -779,24 +785,23 @@ bool GroupBy::isUnsupportedAggregate(sparqlExpression::SparqlExpression* expr) {
   if (!expr->isAggregate()) return false;
 
   // `expr` is an unsupported aggregate
-  if (hasType<SumExpression>(expr) || hasType<MinExpression>(expr) ||
-      hasType<MaxExpression>(expr) || hasType<CountExpression>(expr) ||
-      hasType<GroupConcatExpression>(expr))
+  if (hasAnyType<SumExpression, MinExpression, MaxExpression, CountExpression,
+                 GroupConcatExpression>(expr))
     return true;
 
-  return false;
+  // `expr` is a distinct aggregate
+  return expr->isDistinct();
 }
 
 // _____________________________________________________________________________
 bool GroupBy::findAggregatesImpl(
     sparqlExpression::SparqlExpression* parent,
     sparqlExpression::SparqlExpression* expr, std::optional<size_t> index,
     std::vector<GroupBy::HashMapAggregateInformation>& info) {
-  if (auto val = hasType<sparqlExpression::AvgExpression>(expr);
-      val.has_value()) {
-    // Do not support distinct aggregates
-    if (val.value()->isDistinct()) return false;
+  // Unsupported aggregates
```

Comment:
If I understand this correctly, 
Then this function can also directly take `ParentAndChildIndex` (or optional thereof) because the parent is `nullptr` if and only if the index is `nullopt`.

Then also we increase the type safety and the code coverage (those missed branches which are yellow sometimes help us).